src/adapter/src/catalog/open/builtin_schema_migration.rs

```diff
@@ -196,6 +196,12 @@ pub(super) async fn run(
     // Sanity check to ensure we're not touching durable state in read-only mode.
     assert_eq!(config.read_only, txn.is_savepoint());
 
+    // Tests may provide a dummy build info that confuses the migration step selection logic. Skip
+    // migrations if we observe this build info.
+    if *build_info == DUMMY_BUILD_INFO {
+        return Ok(MigrationResult::default());
+    }
```

Comment:
I do feel like the dummy build info has become more painful as we hardcode more versions into the codebase... there was some similarly awkward working-around in Persist.

In this case it might be possible to make this code work with a stale version here, by eg. filtering the list of migrations passed to `migration.run`. But it's possible in the long run we just want to get rid of the dummy build info entirely...

Reply:
Yeah, my assumption here is that it's fine for tests that don't care about the specific version to not get migrations. If you want to test the migrations, you also have to be explicit about versions. Filtering the migrations has the same effect (there are no migrations before version `0.0.0-dummy`) but prevents us from having the sanity check that looks for migrations at versions larger than the current one. Though if we move the sanity checks into a unit test, I think we can do the filtering and remove the special case for `DUMMY_BUILD_INFO`.

And I'm also not sure why `DUMMY_BUILD_INFO` is needed in the first place. Seems easy to use the actual build info instead. Except perhaps for golden tests that compare to a static expected output.